src/reformat/libyuv.rs

```diff
@@ -130,243 +130,202 @@ impl ConversionFunction {
-                        I422ToARGBMatrixFilter,
-                    ))
-                }
-                PixelFormat::Yuv420 => {
-                    return Some(ConversionFunction::YUVToRGBMatrixFilter(
-                        I420ToARGBMatrixFilter,
-                    ))
-                }
-                _ => {}
-            },
-            _ => {}
+
+        (true, _, Format::Rgba | Format::Bgra, PixelFormat::Yuv422)
+            if rgb.chroma_upsampling.bilinear_or_better_filter_allowed() =>
+        {
+            Some(ConversionFunction::YUVAToRGBMatrixFilter(
+                I422AlphaToARGBMatrixFilter,
+            ))
         }
-        if !rgb.chroma_upsampling.nearest_neighbor_filter_allowed() {
-            return None;
+        (true, _, Format::Rgba | Format::Bgra, PixelFormat::Yuv420)
+            if rgb.chroma_upsampling.bilinear_or_better_filter_allowed() =>
+        {
+            Some(ConversionFunction::YUVAToRGBMatrixFilter(
+                I420AlphaToARGBMatrixFilter,
+            ))
         }
-    }
-    if alpha_preferred {
-        match rgb.format {
-            Format::Rgba | Format::Bgra => match yuv_format {
-                PixelFormat::Yuv444 => {
-                    return Some(ConversionFunction::YUVAToRGBMatrix(I444AlphaToARGBMatrix))
-                }
-                PixelFormat::Yuv422 => {
-                    return Some(ConversionFunction::YUVAToRGBMatrix(I422AlphaToARGBMatrix))
-                }
-                PixelFormat::Yuv420 => {
-                    return Some(ConversionFunction::YUVAToRGBMatrix(I420AlphaToARGBMatrix))
-                }
-                _ => {}
-            },
-            _ => {}
+
+        (_, _, Format::Rgb | Format::Bgr, PixelFormat::Yuv422)
+            if rgb.chroma_upsampling.bilinear_or_better_filter_allowed() =>
+        {
+            Some(ConversionFunction::YUVToRGBMatrixFilter(
+                I422ToRGB24MatrixFilter,
+            ))
         }
+        (_, _, Format::Rgb | Format::Bgr, PixelFormat::Yuv420)
+            if rgb.chroma_upsampling.bilinear_or_better_filter_allowed() =>
+        {
+            Some(ConversionFunction::YUVToRGBMatrixFilter(
+                I420ToRGB24MatrixFilter,
+            ))
+        }
+        (_, _, Format::Rgba | Format::Bgra, PixelFormat::Yuv422)
+            if rgb.chroma_upsampling.bilinear_or_better_filter_allowed() =>
+        {
+            Some(ConversionFunction::YUVToRGBMatrixFilter(
+                I422ToARGBMatrixFilter,
+            ))
+        }
+        (_, _, Format::Rgba | Format::Bgra, PixelFormat::Yuv420)
+            if rgb.chroma_upsampling.bilinear_or_better_filter_allowed() =>
+        {
+            Some(ConversionFunction::YUVToRGBMatrixFilter(
+                I420ToARGBMatrixFilter,
+            ))
+        }
+
+        (true, _, Format::Rgba | Format::Bgra, PixelFormat::Yuv444)
+            if rgb.chroma_upsampling.nearest_neighbor_filter_allowed() =>
```

Comment:
this if guard is not necessary for this case. since yuv format is 444, filter should have no effect.

ditto below on a few other cases of yuv format being 444.